src/main/java/uk/gov/hmcts/reform/ccd/documentam/client/dmstore/DocumentStoreClient.java

```diff
@@ -0,0 +1,181 @@
+                throw new ResourceNotFoundException(
+                    String.format("%s %s", RESOURCE_NOT_FOUND, documentId),
+                    exception
+                );
+            }
+
+            throw new ServiceException(
+                String.format(EXCEPTION_ERROR_ON_DOCUMENT_MESSAGE, documentId),
+                exception
+            );
+        }
+    }
+
+    public void deleteDocument(final UUID documentId, final Boolean permanent) {
+        try {
+            restTemplate.delete(String.format(
+                "%s/documents/%s?permanent=%s",
+                applicationParams.getDocumentURL(),
+                documentId,
+                permanent
+            ));
+        } catch (HttpClientErrorException exception) {
+            handleException(exception, documentId.toString());
+        }
+    }
+
+    public Optional<PatchDocumentResponse> patchDocument(final UUID documentId, final DmTtlRequest dmTtlRequest) {
+        try {
+            final PatchDocumentResponse patchDocumentResponse = restTemplate.patchForObject(
+                String.format("%s/documents/%s", applicationParams.getDocumentURL(), documentId),
+                dmTtlRequest,
+                PatchDocumentResponse.class
+            );
+
+            return Optional.ofNullable(patchDocumentResponse);
+        } catch (HttpClientErrorException exception) {
+            if (HttpStatus.NOT_FOUND.equals(exception.getStatusCode())) {
+                return Optional.empty();
+            }
+            throw new ServiceException(String.format(EXCEPTION_ERROR_ON_DOCUMENT_MESSAGE, documentId), exception);
+        }
+    }
+
+    public void patchDocumentMetadata(final UpdateDocumentsCommand updateDocumentsCommand) {
+        try {
+            restTemplate.patchForObject(
+                String.format("%s/documents", applicationParams.getDocumentURL()),
+                updateDocumentsCommand,
+                Void.class
+            );
+        } catch (HttpClientErrorException exception) {
+            throw new ServiceException(Constants.EXCEPTION_ERROR_MESSAGE, exception);
+        }
+    }
+
+    public DmUploadResponse uploadDocuments(final DocumentUploadRequest documentUploadRequest) {
+        try {
+            return restTemplate.postForObject(
+                String.format("%s/documents", applicationParams.getDocumentURL()),
+                prepareRequestForUpload(documentUploadRequest),
+                DmUploadResponse.class
+            );
+        } catch (HttpClientErrorException exception) {
+            throw new ServiceException(Constants.EXCEPTION_ERROR_MESSAGE, exception);
+        }
+    }
+
+    private HttpEntity<LinkedMultiValueMap<String, Object>> prepareRequestForUpload(final DocumentUploadRequest
+                                                                                        documentUploadRequest) {
+        LinkedMultiValueMap<String, Object> bodyMap = new LinkedMultiValueMap<>();
```

Comment:
Declare type to be the interface ? `MultiValueMap<String, Object>` ?

Reply:
The upload method is going to be rebuilt from the ground up, thus left as is